spyder/plugins/pylint/widgets/pylintgui.py

```diff
@@ -310,14 +337,29 @@ def analyze_button_handler(self):
         """Try to start code analysis when Analyze button pressed."""
         self.start_analysis.emit()
 
+    def get_pylintrc_path(self, filename):
+        """Get the path to the most proximate pylintrc config to the file."""
+        parent = self.parentWidget()
+        if parent is not None:
+            project_dir = parent.main.projects.get_active_project_path()
+        else:
+            project_dir = None
+        search_paths = [
+            osp.dirname(filename),  # File's directory
+            getcwd_or_home(),  # Working directory
+            project_dir,  # Project directory
+            osp.expanduser("~"),  # Home directory

```

Comment:
I think the home path isn't needed here since the `get_pylintrc_path` function ignores it, right? or maybe should be passed as the `home_path` kwarg?

Reply:
Initially I removed it here when I added the `home_path` argument, but having it here makes it more clear that its on the search path and makes the operation of `find_pylintrc_path()` more clear and explicit to the reader, falling back to the base `.pylintrc` in the global user dirs, env vars and etc instead of the user having to infer this behavior if the a .pylintrc isn't found on the project path, as well as making the resulting tests simpler and more consistent.